CMake/FlagsOverride.cmake

```diff
@@ -0,0 +1,7 @@
+if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  foreach(f CMAKE_C_FLAGS_DEBUG CMAKE_CXX_FLAGS_DEBUG CMAKE_C_FLAGS_RELWITHDEBINFO CMAKE_CXX_FLAGS_RELWITHDEBINFO)
+	  if("${${f}}" MATCHES "/Zi")
+		  string(REGEX REPLACE "/Zi" "/Z7" "${f}" "${${f}}")
```

Comment:
No caching tool can save the debug symbols associated with a compilation unit if they are all put in a single PDB file (/Zi). Using /Z7, they are stored in each obj file instead (and then later merged properly by the linker as usual). They will refuse to cache those compilations, so we override the default to use /Z7 instead. It has the potential to reduce contention on that one single file too.